spec/models/manageiq/providers/autosde/storage_manager/refresher_spec.rb

```diff
@@ -31,6 +30,64 @@
       end
     end
 
+    context "targeted refresh" do
+      before { VCR.use_cassette("ems_refresh") { described_class.refresh([ems]) } }
+      let(:system_type_api) { double("SystemTypeApi") }
+      let(:storage_system_api) { double("StorageSystemApi") }
+
+      it "with no targets" do
+        assert_inventory_not_changed { run_targeted_refresh }
+      end
+
+      it "with an existing physical_storage target" do
+        expect(storage_system_api)
+          .to receive(:storage_systems_get)
+          .and_return(
+            [
+              AutosdeOpenapiClient::StorageSystem.new(
+                :auto_add_pools  => true,
+                :component_state => "PENDING_CREATION",
+                :management_ip   => "9.151.159.178",
+                :name            => "9.151.159.178",
+                :status          => "ONLINE",
+                :storage_family  => "ontap_7mode",
+                :system_type     => AutosdeOpenapiClient::SystemType.new(
+                  :component_state => "PENDING_CREATION",
+                  :name            => "FlashSystems/SVC",
+                  :short_version   => "11",
+                  :uuid            => "053446df-ed2b-4822-b9c5-386e85198519",
+                  :version         => "1.2"
+                ),
+                :uuid            => "980f3ceb-c599-49c4-9db3-fdc793cb8666"
+              )
+            ]
+          )
+
+        assert_inventory_not_changed { run_targeted_refresh(ems.physical_storages.first) }
+      end
+
+      def run_targeted_refresh(targets = [])
+        target = InventoryRefresh::TargetCollection.new(:manager => ems, :targets => Array(targets))
+
+        persister = ManageIQ::Providers::Autosde::Inventory::Persister::TargetCollection.new(ems)
+        collector = ManageIQ::Providers::Autosde::Inventory::Collector::TargetCollection.new(ems, target)
+        parser    = ManageIQ::Providers::Autosde::Inventory::Parser::StorageManager.new
```

Comment:
NOTE I went this route instead of `described_class.refresh(target)` because this gives us more control over stubbing the autosde_client because we have access to the collector instance.